enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,641 @@
+     * hasCpuCoresForRole: [success, failure, pending, error]
+     * hasMemoryForRole: [success, failure, pending, error]
+     * isHostnameValid: [success, failure, pending, error]
+     * isHostnameUnique: [success, failure, pending, error]
+     * isPlatformValid: [success, failure, pending, error]
+   * network:
+     * isConnected: [success, failure, pending, error]
+     * isMachineCidrDefined: [success, failure, pending, error]
+     * belongsToMachineCidr: [success, failure, pending, error]
+     * isApiVipConnected: [success, failure, pending, error]
+     * belongsToMajorityGroup: [success, failure, pending, error]
+     * isNtpSynced: [success, failure, pending, error]
+  * progress:
+    * currentStage: enum
+    * progressInfo: string
+    * stageStartedAt: string
+    * stageUpdatedAt: string
+ * connectivity: array of
+   * host
+   * l2Connectivity
+     * outgoingNic: string
+     * outgoingIpAddress: string
+     * remoteIpAddress: string
+     * remoteMac: string
+     * successful: boolean
+   * l3Connectivity
+     * outgoingNic: string
+     * remoteIpAddress: string
+     * successful: boolean
+ * apiVipConnectivity: boolean
+ * ntpSources: array of
+   * sourceName: string
+   * sourceState: [synced, combined, not_combined, error, variable, unreachable]
+
+
+#### REST API Access
+Some REST APIs need to be exposed in addition to the Kubernetes-native APIs
+described below.
+* Download ISO and PXE artifacts: These files must be available for download
+via HTTP, either directly by users or by a BMC.  Because BMCs do not pass
+authentication headers, the Assisted Service must generate some random URL or
+query parameter so that the ISO’s location isn’t easily guessable.
+* Agent APIs (near-term): Until a point where the agent creates and modifies
+AgentDrivenHostInstallation CRs itself, the agent will continue communicating
+with the service via REST APIs.  Currently the service embeds the user’s pull
+secret in the discovery ISO which the agent passes in an authentication
+header.  In this case the service can generate and embed some token which it
+can later validate.
+
+
+#### Hive Integration
+
+Hive has a [ClusterDeployment
+CRD](https://github.com/openshift/hive/blob/master/docs/using-hive.md#clusterdeployment)
+resource that represents a cluster to be created. It includes a "platform"
+section where platform-specific details can be captured. For an agent-based
+workflow, this section will include whatever information is necessary to get an
+ISO and boot hosts.
+
+A new field in the ClusterDeployment spec will optionally reference an
+AgentDrivenClusterInstallation resource, which describes everything that the
+agent-based install tooling needs to know about the cluster that it will be
+creating. Some or all of that new resource may be folded into ClusterDeployment
+in the future, but in the immediate term, keeping it separate will enable much
+quicker development.
```

Comment:
In general once an API goes out, it's near impossible to change it. So before this becomes a real thing, we'll want to transition to the permanent style. During initial devel it makes sense to keep it separate.